File: examples/python/search_example.py
#!/usr/bin/env python3
"""
Notorious API - Full-text/fuzzy search example (Python stdlib only).

Usage:
    python search_example.py "search text"
"""

import json
import sys
import urllib.error
import urllib.parse
import urllib.request

API_URL = "http://localhost:4000"
API_TOKEN = "ntr_your_api_token_here"
WORKSPACE_ID = "your-workspace-id"


def main() -> None:
    if len(sys.argv) != 2:
        raise SystemExit(f"Usage: python {sys.argv[0]} <search text>")
    query = sys.argv[1]

    params = urllib.parse.urlencode({"q": query})
    request = urllib.request.Request(
        f"{API_URL}/api/v1/workspaces/{WORKSPACE_ID}/search?{params}",
        headers={"Authorization": f"Bearer {API_TOKEN}"},
    )
    try:
        with urllib.request.urlopen(request) as response:
            results = json.loads(response.read())
    except urllib.error.HTTPError as error:
        body = json.loads(error.read())
        raise SystemExit(f"Request failed ({error.code}): {body.get('message')}")

    print(f"{len(results)} result(s) for '{query}':")
    for obj in results:
        print(f"  - {obj['title']}  (id: {obj['id']})")


if __name__ == "__main__":
    main()

File: examples/php/search_example.php
<?php
/**
 * Notorious API - Full-text/fuzzy search example (PHP curl extension only).
 *
 * Usage:
 *   php search_example.php "search text"
 */

const API_URL = 'http://localhost:4000';
const API_TOKEN = 'ntr_your_api_token_here';
const WORKSPACE_ID = 'your-workspace-id';

function main(array $argv): void
{
    if (count($argv) !== 2) {
        fwrite(STDERR, "Usage: php {$argv[0]} <search text>\n");
        exit(1);
    }
    $query = $argv[1];

    $path = '/api/v1/workspaces/' . WORKSPACE_ID . '/search?' . http_build_query(['q' => $query]);
    $ch = curl_init(API_URL . $path);
    curl_setopt_array($ch, [
        CURLOPT_RETURNTRANSFER => true,
        CURLOPT_HTTPHEADER => ['Authorization: Bearer ' . API_TOKEN],
    ]);
    $responseBody = curl_exec($ch);
    $status = curl_getinfo($ch, CURLINFO_HTTP_CODE);
    curl_close($ch);

    $data = json_decode($responseBody, true);
    if ($status >= 400) {
        fwrite(STDERR, "Request failed ({$status}): {$data['message']}\n");
        exit(1);
    }

    echo count($data) . " result(s) for '{$query}':\n";
    foreach ($data as $object) {
        echo "  - {$object['title']}  (id: {$object['id']})\n";
    }
}

main($argv);
